app/javascript/flavours/glitch/components/status_content.js

```diff
@@ -301,6 +303,27 @@ class StatusContent extends React.PureComponent {
     this.contentsNode = c;
   }
 
+  /**
+   * Highlights code in code tags.\
+   * Uses highlight.js to convert content inside code tags to span elements with class attributes
+   * @param {String} content - String containing html code tags
+   * @returns content with highlighted code inside code tags, or content if not found
+   */
+  highlightCode (content) {
+    // RegEx matches every code tag or block and replaces them with highlighted code if lang defined
+    return content.replace(/((?:<pre>)?<code(?: title="(\w+)?")?>)(.+?)(<\/code>(?:<\/pre>)?)/g, (match, startTags, lang, code, endTags) => {
```

Comment:
I think this should operate on the DOM rather than using regexp-based replacement.

Reply:
I think I tried it initially, but couldn't get it to work.
I can double check though.
Thanks for reviewing this!

Reply:
Ah yes, the highlighter really doesn't like DOM, but I made it work.
Is this what you had in mind?

```js
/**
 * Highlights code in code tags.\
 * Uses highlight.js to convert content inside code tags to span elements with class attributes
 * @param {String} html - string containing html code tags
 * @returns `html` with highlighted code inside code tags
 */
export const highlightCode = (html) => {
  // highlightJS complains when unescaped html is given
  highlightjs.configure({ ignoreUnescapedHTML: true });

  const wrapper = document.createElement('div');
  wrapper.innerHTML = html;
  let elements = wrapper.querySelectorAll('code');

  elements.forEach((value) => {
    // Check if title is a valid language
    if (highlightjs.getLanguage(value.getAttribute('title')) !== undefined) {
      // Set title as class attribute, since highlightElement cannot be given a language
      // highlightJS will read this attribute and use it to highlight in the proper language
      value.setAttribute('class', value.getAttribute('title'));

      // Replace <br> as highlighter removes them, messing up formatting
      value.innerHTML = value.innerHTML.replace(/<br\s*\/?>/g, '\n');

      // Highlight the code element
      highlightjs.highlightElement(value);

      // highlightJS adds own class attribute, remove it again to not mess up styling
      value.removeAttribute('class');
    }
  });

  return wrapper.innerHTML;
};
```
(I moved it to `utils/html.js` more or less by accident, but since this wouldn't be needed anywhere else, it can be easily moved back to `status_content.js`)

Reply:
I'd also replace the `br` tags using the DOM I guess, but this is much better!

Reply:
I... have no idea how that would work tbh

Reply:
You'd get the list with `value.getElementsByTagName('br')` and replace individual elements with something like `o.replaceWith('')`, though you must take care of not replacing while iterating (first store the list to an array or something)

Reply:
Like this?
```js
// Replace <br> as highlightJS removes them, messing up formatting
let brTags = value.getElementsByTagName('br');
for (let br of brTags) {
  br.replaceWith('\n');
}
```

Reply:
No, this wouldn't work because `brTags` is an iterator that isn't safe to use when you're mutating the contents. You could use `Array.from(…)`

Reply:
Like this?
```js
// Replace <br> as highlightJS removes them, messing up formatting
let brTags = Array.from(value.getElementsByTagName('br'));
for (let br of brTags) {
  br.replaceWith('\n');
}
```

Reply:
Yeah this should work